### ISSUE TITLE
K8SSAND-1113 ⁃ PVC is not used when deploying with the example config

### ISSUE THREAD

**Issue by Yuyan-Li** [k8ssandra/cass-operator#247]
**What happened?**

After creating a Cassandra cluster, everything is created as expected. But the data is not written into the PVCs. Inside the container I can see that the data is actually written to `/opt/cassandra/data/data`. This is because the default location is `$CASSANDRA_HOME/data/data` (can be seen [here](https://github.com/apache/cassandra/blob/9fe62b3e40147fda2cc081744bd375b04574aef7/conf/cassandra.yaml#L189)).

To write to the PVC one has to add these lines to the `cassandra.yaml`.

```
data_file_directories:
    - /var/lib/cassandra/data
hints_directory: /var/lib/cassandra/hints
commitlog_directory: /var/lib/cassandra/commitlog
cdc_raw_directory: /var/lib/cassandra/cdc_raw
saved_caches_directory: /var/lib/cassandra/saved_caches
```

**Did you expect to see something different?**
I expected the cass-operator to set all configs so that data is automatically written to the VolumeClaim. If that expectation was wrong then I would expect the example configuration to reflect that.

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**

* Cass Operator version:
cass-operator v1.5.0
* Kubernetes version information:
1.20
* Kubernetes cluster kind:
Openshift
* Manifests:

```
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
metadata:
  name: dc1
spec:
  clusterName: a-cassandra-cluster
  serverType: cassandra
  serverVersion: 3.11.7
  managementApiAuth:
    insecure: {}
  size: 3
  resources:
    requests:
      memory: 5Gi
      cpu: 3
    limits:
      memory: 5Gi
      cpu: 3
  storageConfig:
    cassandraDataVolumeClaimSpec:
      storageClassName: storage
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 100Gi
  config:
    cassandra-yaml:
      num_tokens: 8
      authenticator: org.apache.cassandra.auth.AllowAllAuthenticator
      authorizer: org.apache.cassandra.auth.AllowAllAuthorizer
      role_manager: org.apache.cassandra.auth.CassandraRoleManager
    jvm-options:
      initial_heap_size: 2G
      max_heap_size: 2G
      additional-jvm-opts:
        - "-Dcassandra.system_distributed_replication_dc_names=dc1"
        - "-Dcassandra.system_distributed_replication_per_dc=3"
```

* Cass Operator Logs:
n/a

**Anything else we need to know?**:

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1113) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1113
┆priority: Medium

**Comment by burmanm** [comment]
I was unable to replicate this behavior with 1.9.0, using https://github.com/k8ssandra/cass-operator/blob/master/tests/testdata/default-single-rack-single-node-addtional-volumes-dc.yaml and modifying it to use 4.0.1 (3.11.10 is a broken version, so I didn't test that one):

The version you're using (1.5.0) is very old.

```console
cassandra@cluster2-dc2-r1-sts-0:/var/lib/cassandra$ ls -ltr
total 0
drwxr-xr-x 2 cassandra cassandra   6 Dec 13 15:00 saved_caches
drwxr-xr-x 2 cassandra cassandra   6 Dec 13 15:00 hints
drwxrwxrwx 2 root      root       80 Dec 13 15:00 commitlog
drwxr-xr-x 7 cassandra cassandra 107 Dec 13 15:01 data
drwxr-xr-x 3 cassandra cassandra  22 Dec 13 15:01 mcac_data
cassandra@cluster2-dc2-r1-sts-0:/var/lib/cassandra$ ls -l data
total 12
drwxr-xr-x 22 cassandra cassandra 4096 Dec 13 15:00 system
drwxr-xr-x  7 cassandra cassandra 4096 Dec 13 15:01 system_auth
drwxr-xr-x  5 cassandra cassandra  181 Dec 13 15:01 system_distributed
drwxr-xr-x 12 cassandra cassandra 4096 Dec 13 15:00 system_schema
drwxr-xr-x  4 cassandra cassandra  102 Dec 13 15:01 system_traces
cassandra@cluster2-dc2-r1-sts-0:/var/lib/cassandra$
```

**Comment by Yuyan-Li** [comment]
Thanks for the info. Unfortunately, I don't have control over the operator version we are using and currently we do not have a newer version available to us.

**Comment by Yuyan-Li** [comment]
For me it looks like this (before changing the config):
```
sh-4.2$ ls -l /opt/cassandra/data/
total 0
drwxr-xr-x. 2 1001050000 root  80 Dec  8 14:25 commitlog
drwxr-xr-x. 9 1001050000 root 152 Dec  8 14:30 data
drwxr-xr-x. 2 1001050000 root   6 Dec  8 14:25 hints
drwxr-xr-x. 3 1001050000 root  22 Dec  8 14:26 mcac_data
drwxr-xr-x. 2 1001050000 root  49 Dec 13 18:25 saved_caches
```
Is the different user relevant? I think that's due to our cluster setup.

For the manifest I followed this one: https://github.com/datastax/cass-operator/blob/master/docs/user/README.md#example-config

**Comment by Yuyan-Li** [comment]
After some more testing, I found out, that the issue only happens with the UBI image of the cass-operator. It happens for both [v1.7.1](https://hub.docker.com/layers/k8ssandra/cass-operator/v1.7.1-ubi7/images/sha256-859e64f8542decca72982f02c240bcb3c37e7cfdf8589223a93e0aad83305b37?context=explore) and [v1.5.0](https://hub.docker.com/layers/datastax/cass-operator/1.5.0-ubi/images/sha256-314b8edeba0f34547cc171c68b767b5a63e3a44132bb80e1730a2204665717f8?context=explore)

**Comment by burmanm** [comment]
The image I used for testing (1.9.0) is UBI always these days. But the build process was changed between 1.7.1 and 1.8.0 for UBI, so it's possible that older images have this issue.

**Comment by Yuyan-Li** [comment]
Thanks for the info. Unfortunately, Redhat does not yet provide a newer image than 1.7.1

**Comment by bradfordcp** [comment]
We'll work on getting the Red Hat catalog updated.

**Comment by chrisob** [comment]
Can confirm that RH catalog lists 1.5.0 as the latest available version, would be great to have 1.9.0 as an option in OpenShift :pray:
